ffi/freetype.lua

```diff
@@ -125,10 +125,46 @@ function FTFace_mt.__index:getHeightAndAscender()
 end
 
 function FTFace_mt.__index:done()
+    -- Note that past this point, accessing self is illegal, FT2 frees the pointer
     assert(ft2.FT_Done_Face(self) == 0, "freetype error when freeing face")
 end
 
-FTFace_mt.__gc = FTFace_mt.__index.done;
+function FTFace_mt.__index:getInfo()
+    local finfo = {
+        name = ffi.string(self.family_name),
+        -- Style
+        mono = bit.band(tonumber(self.face_flags), ft2.FT_FACE_FLAG_FIXED_WIDTH) ~= 0,
+        hint = bit.band(tonumber(self.face_flags), ft2.FT_FACE_FLAG_HINTER) ~= 0,
+        bold = bit.band(tonumber(self.style_flags), ft2.FT_STYLE_FLAG_BOLD) ~= 0,
+        italic = bit.band(tonumber(self.style_flags), ft2.FT_STYLE_FLAG_ITALIC) ~= 0,
+        serif = nil,
+        ui = false,
+        names = nil,
+    }
+
+    -- In practice, just going by latin name can tell us more than the depressingly absent tags
+    finfo.ui = finfo.name:match("UI$") ~= nil
+    local lname = finfo.name:lower()
+    if lname:match("serif") then
+        finfo.serif = true
+    elseif lname:match("sans") then
+        finfo.serif = false
+    elseif bit.band(tonumber(self.face_flags), ft2.FT_FACE_FLAG_SFNT) ~= 0 then
+        local os2 = ft2.FT_Get_Sfnt_Table(self, ft2.FT_SFNT_OS2)
+        if os2 ~= nil then
+            local kls = tonumber(ffi.cast("TT_OS2*", os2).sFamilyClass)
+            if kls ~= 0 then -- 0 is usually bogus
+                -- class 8 = sans, otherwise serif
+                finfo.serif = bit.band(kls, 0x0800) == 0
```

Comment:
Mhhh, in crengine, I killed the loosy support for serif/sans as it looked we couldn't get that info :)
I'll pretend I didn't see that :) (and it would be painful as we'd need the user to choose a main font for serif and another for sans - and I don't really care)

Reply:
its not really all that accurate (panose tells a bit more for *some* fonts), but it's still 50/50, with heavy preference for serif, so eh.